http/document_service.go

```diff
@@ -260,46 +270,145 @@ func decodeGetDocumentsRequest(ctx context.Context, r *http.Request) (*getDocume
 	}, nil
 }
 
-// handleGetDocument is the HTTP handler for the GET /api/v2/documents/:ns/:id route.
-func (h *DocumentHandler) handleGetDocument(w http.ResponseWriter, r *http.Request) {
+func (h *DocumentHandler) handlePostDocumentLabel(w http.ResponseWriter, r *http.Request) {
 	ctx := r.Context()
+	d, _, s, a, err := h.getDocument(w, r)
+	if err != nil {
+		EncodeError(ctx, err, w)
+		return
+	}
+	req, err := decodePostLabelMappingRequest(ctx, r, influxdb.DocumentsResourceType)
+	if err != nil {
+		EncodeError(ctx, err, w)
+		return
+	}
 
-	req, err := decodeGetDocumentRequest(ctx, r)
+	if err := req.Mapping.Validate(); err != nil {
+		EncodeError(ctx, err, w)
+		return
+	}
+
+	if err := h.LabelService.CreateLabelMapping(ctx, &req.Mapping); err != nil {
+		EncodeError(ctx, err, w)
+		return
+	}
+
+	label, err := h.LabelService.FindLabelByID(ctx, req.Mapping.LabelID)
 	if err != nil {
 		EncodeError(ctx, err, w)
 		return
 	}
+	d.Labels = append(d.Labels, label)
```

Comment:
I think Labels on the document struct are intended to be read-only and stored in the labels service, not as part of the document.